test/Conversion/ExportAIGER/binary.mlir

```diff
@@ -0,0 +1,25 @@
+// RUN: circt-translate --export-aiger %s -o %t.aig --verify-diagnostics
+// RUN: cat %t.aig | FileCheck %s --check-prefix=BINARY
+// RUN: circt-translate --import-aiger %t.aig | FileCheck %s
+
+// Test complex binary encoding with proper ordering
+// BINARY:      aig 9 3 2 2 4
+hw.module @mixed_logic(in %a: i1, in %b: i1, in %c: i1, out x: i1, out y: i1, in %clk: !seq.clock) {
+  %comb_logic = aig.and_inv %a, %b : i1
+  %next_x = aig.and_inv %c, %comb_logic : i1
+  %next_y = aig.and_inv %reg_x, %reg_y : i1
+  %reg_x = seq.compreg %next_x, %clk : i1
+  %reg_y = seq.compreg %next_y, %clk : i1
+  %final_out = aig.and_inv %next_x, %next_y : i1
+  hw.output %final_out, %reg_y : i1, i1
+}
+
+// CHECK-LABEL:  hw.module @aiger_top(in %a : i1, in %b : i1, in %c : i1, out x : i1, out y : i1, in %clock : !seq.clock) {
+// CHECK:         %reg_x = seq.compreg %[[AND_INV_2:.+]], %clock : i1  
+// CHECK-NEXT:    %reg_y = seq.compreg %[[AND_INV_3:.+]], %clock : i1  
+// CHECK-NEXT:    %[[AND_INV_1:.+]] = aig.and_inv %b, %a : i1
+// CHECK-NEXT:    %[[AND_INV_2]] = aig.and_inv %[[AND_INV_1]], %c : i1
+// CHECK-NEXT:    %[[AND_INV_3:.+]] = aig.and_inv %reg_y, %reg_x : i1
+// CHECK-NEXT:    %[[AND_INV_4:.+]] = aig.and_inv %[[AND_INV_3]], %[[AND_INV_2]] : i1
+// CHECK-NEXT:    hw.output %[[AND_INV_4]], %reg_y : i1, i1
+// CHECK-NEXT: }
```

Comment:
Super cool that we can round-trip through AIGER!

I'm wondering if something like `dbg.variable` could be used to capture additional names :thinking:. It wouldn't do anything here because names like `%next_x` get dropped anyway, but it might be an interesting way to preserve a tiny bit of debug info in the AIGER file sometime down the road :grin:

Reply:
By design ExportAIGER fails if there is an unknown operaton like dbg.variable. We could add an additional output for dbg.variable but we need ABI-ish information for that. AIGERRunner pass indeed is meant to handle such unknown operations so dbg.variable will should compose well.